agent/GCEWindowsAgent/AddressReader.cs

```diff
@@ -46,26 +48,25 @@ private List<IPAddress> ParseForwardIpsResult(string[] forwardedIps)
             return addresses;
         }
 
-        public List<IPAddress> GetMetadata(MetadataJson metadata)
+        public Dictionary<PhysicalAddress, List<IPAddress>> GetMetadata(MetadataJson metadata)
         {
-            try
-            {
-                string[] forwardedIps = metadata.Instance.NetworkInterfaces[0].ForwardedIps;
-                return ParseForwardIpsResult(forwardedIps);
-            }
-            catch (NullReferenceException)
+            Dictionary<PhysicalAddress, List<IPAddress>> data = new Dictionary<PhysicalAddress, List<IPAddress>>();
+
+            foreach (NetworkInterfacesJson entry in metadata.Instance.NetworkInterfaces)
             {
-                return null;
+                PhysicalAddress mac = PhysicalAddress.Parse(entry.MAC);
+                data.Add(mac, ParseForwardIpsResult(entry.ForwardedIps));
             }
+            return data;
         }
 
-        public bool CompareMetadata(List<IPAddress> oldMetadata, List<IPAddress> newMetadata)
+        public bool CompareMetadata(Dictionary<PhysicalAddress, List<IPAddress>> oldMetadata, Dictionary<PhysicalAddress, List<IPAddress>> newMetadata)
```

Comment:
If I recall correctly, there is a 100 character limit (or 110?). Either way, this is exceeding it.